validator/lib/validateTileset.js

```diff
@@ -28,7 +28,12 @@ function validateTileset(tileset, tilesetDirectory) {
         return Promise.resolve(message);
     }
 
-    return validateTileHierarchy(tileset.root, tilesetDirectory);
+    message = validateTileHierarchy(tileset.root, tilesetDirectory);
+    if (defined(message)) {
+        return Promise.resolve(message);
+    }
+
+    return undefined;
```

Comment:
`Promise.resolve(validateTileHierarchy(tileset.root, tilesetDirectory));` is equivalent here.